src/Orleans/Streams/PersistentStreams/PersistentStreamProvider.cs

```diff
@@ -60,11 +60,18 @@ public GrainFactoryServiceProvider(IStreamProviderRuntime providerRuntime)
             }
             public object GetService(Type serviceType)
             {
-                if (serviceType == typeof (GrainFactory))
+                var service = providerRuntime.ServiceProvider?.GetService(serviceType);
```

Comment:
Just double checked.  Runtime is properly checked at provider initialization.  As a private internal class this service provider can trust that and does not need to check for itself.